frame/support/procedural/src/pallet/expand/error.rs

```diff
@@ -118,18 +150,40 @@ pub fn expand_error(def: &mut Def) -> proc_macro2::TokenStream {
 			#config_where_clause
 		{
 			fn from(err: #error_ident<#type_use_gen>) -> Self {
+				// Ensure that we can still use `try_into` in earlier editions
+				#[cfg(not(feature = "prelude_2021"))]
+				use core::convert::TryInto;
+				use #frame_support::codec::Encode;
 				let index = <
 					<T as #frame_system::Config>::PalletInfo
 					as #frame_support::traits::PalletInfo
 				>::index::<Pallet<#type_use_gen>>()
 					.expect("Every active module has an index in the runtime; qed") as u8;
+				let mut encoded = err.encode();
+				encoded.resize(#frame_support::MAX_PALLET_ERROR_ENCODED_SIZE, 0);
 
 				#frame_support::sp_runtime::DispatchError::Module(#frame_support::sp_runtime::ModuleError {
 					index,
-					error: err.as_u8(),
+					error: encoded.try_into().expect("encoded error is resized to be equal to 4 bytes; qed"),
 					message: Some(err.as_str()),
 				})
 			}
 		}
+
+		#[macro_export]
```

Comment:
Same code as in line 39. Maybe we could just generate it once at the top of the function?

Reply:
There's a slight difference in what goes inside of the `tt_return` macro -- this piece of code here sends the error type name as an argument back to its caller.